### DISCUSSION TITLE
dynamic docstring not picked up by intellisense

### DISCUSSION THREAD

**Discussion by koubaa** [microsoft/pylance-release#2676] [Enhancement]
### VS Code version

1.61.2

### Extension version

v2021.11.1422169775

### OS type

Windows

### OS version

10

### Python distribution

python.org

### Python version

3.9

### Language server

Default

### Expected behaviour

1. Have a python package that uses dynamic docstrings, for example a function like this:
```python
def foo(arg1, arg2):
    return None

foo.doc = "hello"
```

2. Install this package to the python interpreter used by VSCode.

3. import this package, and start to call the function. You will see the function in the auto-comoplete list and you see "hello" when you select it.

### Actual behaviour

1. Have a python package that uses dynamic docstrings, for example a function like this:
```python
def foo(arg1, arg2):
    return None

foo.doc = "hello"
```
2. Install this package to the python interpreter used by VSCode.

3. import this package, and start to call the function. You will see the function in the auto-comoplete list and you don't see any docstring

### Steps to reproduce

1. Have a python package that uses dynamic docstrings, for example a function like this:

```python
def foo(arg1, arg2):
    return None

foo.doc = "hello"
```

2. Install this package to the python interpreter used by VSCode.

3. import this package, and start to call the function. You will see the function in the aut0-comoplete list and you don't see a docstring

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by judej** [comment]
Thanks for the suggestion. Currently we do not support dynamic docstrings.

**Comment by Malachov** [comment]
Any expectations of future behavior? I don't know whether it's taken as a bad practice or it's too complicated to implement for some reason or it could be implemented in the future..

As it works on sphinx docs, it would be nice feature (though it can be just my opinion - I don't know how about other users).

I read somewhere, that docstrings wont be dynamic in future versions (f-strings or some alternative), so solve it within IDE seems legit to me.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Reply by raceychan** [reply]
This feature could be widely used by many developers, a simple use case would be decorators that rewrite functions docstring.
for example:
```python
def deprecated(func):
    doc = f'{func.__name__} will be deprecated in next version'
    def wrapper(*args, **kwargs):
        return func(*args, **kwargs)
    wrapper.__doc__ = doc
    return wrapper
```

**Reply by nineteendo** [reply]
Could also be useful for generated functions:
```python
def make_print(value: str):
    """Make print function."""
    def print_func() -> None:
        print(value)

    print_func.__doc__ = f'Print {value}'
    return print_func


print_hello = make_print('Hello!')
print_world = make_print('World!')

print_hello()
print_world()
```

**Reply by Viicos** [reply]
I'm not sure how this would be possible: pylance does not execute any runtime code, so it will not be able to support such dynamic decorators
